strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/repositories/ArtifactIdGroupRepository.java

```diff
@@ -0,0 +1,226 @@
+    {
+        return queries.findMatching(storageId, repositoryId, page);
+    }
+
+    public Optional<ArtifactIdGroup> findAllArtifactsInGroup(String storageId,
+                                                             String repositoryId,
+                                                             String artifactId)
+    {
+        return findArtifactsGroupWithTag(storageId, repositoryId, artifactId, Optional.empty());
+    }
+
+    public Optional<ArtifactIdGroup> findArtifactsGroupWithTag(String storageId,
+                                                               String repositoryId,
+                                                               String artifactId,
+                                                               Optional<ArtifactTag> tag)
+    {
+        org.carlspring.strongbox.storage.repository.Repository repository = configurationManager.getRepository(storageId, repositoryId);
+        
+        ArtifactIdGroup artifactIdGroup = new ArtifactIdGroupEntity(storageId, repositoryId, artifactId);
+        EntityTraversal<Vertex, ArtifactIdGroup> t = g().V()
+                                                        .hasLabel(Vertices.ARTIFACT_ID_GROUP)
+                                                        .has("uuid", artifactIdGroup.getUuid())
+                                                        .map(adapter.fold(Optional.ofNullable(repository)
+                                                                                  .map(org.carlspring.strongbox.storage.repository.Repository::getLayout)
+                                                                                  .map(ArtifactLayoutLocator.getLayoutByNameEntityMap()::get)
+                                                                                  .map(ArtifactLayoutDescription::getArtifactCoordinatesClass),
+                                                                          tag));
+        if (!t.hasNext())
+        {
+            return Optional.empty();
+        }
+
+        return Optional.of(t.next());
+    }
+
+    public Boolean artifactsExists(Set<String> storageRepositoryIds,
+                                   String artifactId,
+                                   Collection<String> coordinateValues)
+    {
+        Set<String> artifactIdGroupIds = storageRepositoryIds.stream()
+                                                             .map(storageRepositoryId -> storageRepositoryId.split(":"))
+                                                             .map(storageRepositoryId -> new ArtifactIdGroupEntity(
+                                                                     storageRepositoryId[0],
+                                                                     storageRepositoryId[1], artifactId))
+                                                             .map(ArtifactIdGroup::getUuid)
+                                                             .collect(Collectors.toSet());
```

Comment:
Cosmetic: Indentation.
```suggestion
        Set<String> artifactIdGroupIds = storageRepositoryIds.stream()
                                                             .map(storageRepositoryId -> storageRepositoryId.split(":"))
                                                             .map(storageRepositoryId -> new ArtifactIdGroupEntity(storageRepositoryId[0],
                                                                                                                   storageRepositoryId[1],
                                                                                                                   artifactId))
                                                             .map(ArtifactIdGroup::getUuid)
                                                             .collect(Collectors.toSet());
```